src/dependency-manager/src/utils/slugs.ts

```diff
@@ -268,3 +288,30 @@ export class GatewaySlugUtils extends SlugUtils {
     };
   };
 }
+
+export class SlugParser {
```

Comment:
@tjhiggins Pertaining to this comment: https://gitlab.com/architect-io/architect-cli/-/issues/157#note_453801763

This is the meat, ignore the other rename changes. We'd want to try an ordered parsing to grab the relevant info from whatever string we were passed (`component`, `account/component`, `account/component:tag`. If David is good with the product changes, I think something like this would work for parsing an unknown slug.